### DISCUSSION TITLE
Update many - without querying

### DISCUSSION THREAD

**Discussion by Dzejkop** [SeaQL/sea-orm#655] [Q&A]
I would like to update multiple entities at the same time without getting them from the database first.

The docs for the update https://www.sea-ql.org/SeaORM/docs/basic-crud/update describe 2 approaches:
1. Fetch & then update
2. Update multiple items, but with no way to specify the values for different entities.

Let's consider the schema
```sql
CREATE TABLE my_table (
    id INTEGER PRIMARY KEY,
    name TEXT NOT NULL,
    age INTEGER NOT NULL
);

INSERT INTO my_table 
VALUES 
  (1, 'John', 20),
  (2, 'Jane', 32),
  (3, 'Michael', 42);
```

Currently I'm running something like
```sql
UPDATE my_table
SET name = tmp.name,
    age = tmp.age
FROM ( 
  VALUES
      (1, 'Whatever', 2), 
      (2, 'Hello', 4), 
      (3, 'World', 4) 
) AS tmp(id, name, age) 
WHERE my_table.id = tmp.id;
```

and I would like the behaviour to remain the same. Is there a way to do this with SeaOrm?

**Comment by billy1624** [comment]
Hey @Dzejkop, sorry for the delay. If you want to construct and execute such complex query, I would suggest you use SeaQuery then execute it with SeaORM.

For example
- https://github.com/SeaQL/starfish-ql/blob/0639a1301ae863a00c6bc4d6f9268b69902f4163/starfish/starfish-core/src/mutate/edge.rs#L71-L90
- https://www.sea-ql.org/SeaORM/docs/basic-crud/raw-sql#execute-query-using-execute-method
